includes/bp-groups/classes/class-bp-rest-group-invites-endpoint.php

```diff
@@ -148,48 +168,160 @@ public function get_items( $request ) {
 	 * @return bool|WP_Error
 	 */
 	public function get_items_permissions_check( $request ) {
-		$retval = true;
+		$allow = true;
+		$user_id        = bp_loggedin_user_id();
+		$group_id_arg   = isset( $request['group_id'] ) ? absint( $request['group_id'] ) : false;
+		$user_id_arg    = isset( $request['user_id'] ) ? absint( $request['user_id'] ) : false;
+		$inviter_id_arg = isset( $request['inviter_id'] ) ? absint( $request['inviter_id'] ) : false;
+
+		// If the query is not restricted by group or user, limit it to the current user, if not an admin.
+		if ( ! $group_id_arg && ! $user_id_arg && ! bp_current_user_can( 'bp_moderate' ) ) {
+			$user_id_arg = $user_id;
+		}
 
-		if ( ! is_user_logged_in() ) {
-			$retval = new WP_Error(
+		if ( ! $user_id ) {
+			$allow = new WP_Error(
 				'bp_rest_authorization_required',
 				__( 'Sorry, you need to be logged in to see the group invitations.', 'buddypress' ),
 				array(
 					'status' => rest_authorization_required_code(),
 				)
 			);
+
+			/**
+			 * Users can see invitations if they
+			 * - are a site admin
+			 * - are a group admin of the subject group (group_id must be specified)
+			 * - are the invite recipient (user_id must be specified)
+			 * - are the inviter (inviter_id must be specified)
+			 * So, the request must be scoped if the user is not a site admin.
+			 */
+		} else if ( bp_current_user_can( 'bp_moderate' )
+				|| $group_id_arg && $this->can_see( $group_id_arg )
+				|| $user_id_arg === $user_id
+				|| $inviter_id_arg === $user_id
+			) {
+			// Do nothing.
+		} else {
+			$allow = new WP_Error(
+				'bp_rest_authorization_required',
+				__( 'Sorry, you are not allowed to fetch group invitations with those arguments.', 'buddypress' ),
+				array(
+					'status' => rest_authorization_required_code(),
```

Comment:
This should be a `500` error code.

```suggestion
					'status' => 500,
```

Reply:
👍